### DISCUSSION TITLE
Any way to run services containers alongside the task container? (Postgres, Redis, etc)

### DISCUSSION THREAD

**Discussion by patmaddox** [concourse/concourse#6938] [Help & Support] [answered]
I am converting some CI scripts from Gitlab CI to Concourse, and I'm wondering how to handle the [services](https://docs.gitlab.com/ee/ci/services/index.html) section. I looked at the [rails example](https://concourse-ci.org/rails-example.html), it installs Postgres directly into the task image. This works, but it leads to very heavy images, and we can't make use of the many pre-built images that exist.

Does Concourse have any concept of running other containers alongside the task container, so we can do things like [run a Postgres container and have the task container connect to it](https://docs.gitlab.com/ee/ci/services/index.html#passing-cicd-variables-to-services)?

**Comment by taylorsilva** [comment, marked as answer]
Right now the current solution is to run docker inside a concourse container: https://blog.concourse-ci.org/running-docker-in-concourse/

Long-term, implementing the services RFC would solve your problem nicely: https://github.com/concourse/rfcs/pull/84

**Reply by martinwilkerson-ons** [reply]
Just to note that the URL for that blog article has changed to https://blog.concourse-ci.org/posts/2020-12-29-running-docker-in-concourse/.
